### ISSUE TITLE
Use our custom monad in servant

### ISSUE THREAD

**Issue by ChShersh** [kowainik/hintman#10]
Blocked by #9 

After #9 is implemented we should introduce our custom monad that stores `Config` in `ReaderT` context and use the config.

**Comment by cronokirby** [comment]
Couldn't we already integrate the existing context into a monad for servant, and then add the Config to that context in another issue?

**Comment by ChShersh** [comment]
@cronokirby Well, I don't think that existing `Context` should be used as application environment. For example, `port` is not useful in the environment. So usually CLI arguments are called `Options`, and then there's function that takes CLI options, configuration file options and combines them all into application `Env`.

**Comment by cronokirby** [comment]
Ok I see

**Comment by ChShersh** [comment]
@nixorn The idea looks legit. You indeed need to use `hoistServerWithContext` function. But instead of `nt` you should implement function like:

```haskell
runAppAsHandler :: Env -> App a -> Handler a
```

And use it instead. Some example from our work:

```haskell
server :: AppEnv -> Server API
server env = hoistServerWithContext
    (Proxy @API)
    (Proxy @AuthContextHandlers)
    (runAppAsHandler env)
    (toServant mpowerServer)

application :: AppEnv -> Application
application env = serveWithContext (Proxy @API) (authServerContext env) (server env)
```

**Comment by nixorn** [comment]
@ChShersh still in trouble https://github.com/kowainik/hintman/blob/10-custom-monad-in-servant/src/Hintman/Server.hs
```
/home/nixorn/work/kowaink/hintman/src/Hintman/Server.hs:51:21: error:
    • Couldn't match kind ‘*’ with ‘[*]’
      When matching types
        context0 :: [*]
        HintmanAuthContext :: *
      Expected type: Proxy context0
        Actual type: Proxy HintmanAuthContext
    • In the second argument of ‘hoistServerWithContext’, namely
        ‘(Proxy @HintmanAuthContext)’
      In the second argument of ‘($)’, namely
        ‘hoistServerWithContext
           (Proxy @HintmanAPI)
           (Proxy @HintmanAuthContext)
           (runAppAsHandler s)
           (toServant hintmanServer)’
      In the expression:
        serveWithContext (Proxy @HintmanAPI) hintmanAuthServerContext
          $ hoistServerWithContext
              (Proxy @HintmanAPI)
              (Proxy @HintmanAuthContext)
              (runAppAsHandler s)
              (toServant hintmanServer)
   |
51 |                    (Proxy @HintmanAuthContext)
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^

```

Can you explain why kinds different here? I can not get what ghc want from me)
